charts/linkerd2/templates/proxy-injector-rbac.yaml

```diff
@@ -61,8 +61,8 @@ metadata:
 type: Opaque
 data:
   {{ $ca := genCA (printf "linkerd-proxy-injector.%s.svc" .Namespace) 365 -}}
-  crt.pem: {{ b64enc $ca.Cert }}
-  key.pem: {{ b64enc $ca.Key }}
+  crt.pem: {{ ternary (b64enc $ca.Cert) (b64enc .ProxyInjector.CrtPEM) (empty .ProxyInjector.CrtPEM) }}
```

Comment:
During upgrade, `.ProxyInjector.CrtPEM` and `.ProxyInjector.KeyPEM` will be assigned the values of the existing cert and key, so that they can be re-used. This change is also added to the service profile validator and tap RBAC templates.

Also, the `.ProxyInjector.CrtPEM` and `.ProxyInjector.KeyPEM` variables are needed to pin the test fixtures in the install and upgrade tests. Otherwise, Helm will always regenerate the cert/key pairs on every test run. If needed, we can revisit whether it's necessary to use Helm to auto-generate the TLS secrets.